Logix/Components/GetComponent.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using BaseX;
+using FrooxEngine;
+
+namespace FrooxEngine.LogiX.Components
+{
+    [NodeName("Get Component")]
+    [Category("LogiX/Components")]
+    public class GetComponent : LogixOperator<Component>
+    {
+        public readonly Input<Slot> Slot;
+        public readonly Input<string> ComponentName;
+
+        public static readonly List<Type> BlacklistedTypes = new List<Type>()
```

Comment:
I'm not particularly amused.
This needs a permission system, not individual types blacklisted. With component-level access, you can basically recreate inspectors and use them without having permissions from the host. This is a huge security vulnerability. Session hosts need the ability to disable component access completely, the asset slot, it's children, and other important world slots need to be blocked completely at all times, and individual users need the ability to block component access from themselves and their avatars.
Permissions also need to apply to all nodes that access components. I see how you attempted to appease me by adding this here, but you added it to literally nothing else, so it doesn't change the safety of this implementation at all.

Reply:
I'm not allowing users to do anything they wouldn't be able to do otherwise. A Guest user can use Get Slot and Destroy Slot to get access to the hierarchy and cause disruptions. Combined with refhacking, there isn't much currently in terms of hard read/write enforcement. I don't see how these additions are opening any security vulnerabilities that weren't preexisting.

My latest commit adds base functionality for protecting Slots and their children from any component-access-related operations. I will add a host-only in-game configurator for modifying blacklisted components, blacklisted slots, and a flag stating whether any component access can be performed in that session.

Reply:
I never said that the base game had no security issues. Both being able to freely destroy slots and refhacking are security issues I would patch myself if I had the power to. Refhacking is already heavily discouraged and, if I recall correctly, will be patched out. "Refhacking already allows you to do this" is not a justification for adding more security vulnerabilities.

If you can make it so that users can block component access on themselves with a config, similar to the session config but still blocking on their user even if component access for the session is enabled, I'll allow this to merge.